clients/lsp-svlangserver.el

```diff
@@ -0,0 +1,137 @@
+  :type 'string
+  :safe (lambda (x) (stringp x)))
+
+(defcustom lsp-clients-svlangserver-module-path "svlangserver.js"
+  "svlangserver module path"
+  :group 'lsp-svlangserver
+  :type 'string
+  :safe (lambda (x) (stringp x)))
+
+(defcustom lsp-clients-svlangserver-workspace-additional-dirs nil
+  "Additional directories to be managed by this instance of svlangserver"
+  :group 'lsp-svlangserver
+  :type '(repeat string)
+  :safe (lambda (x) (seq-every-p #'stringp x)))
+
+(defcustom lsp-clients-svlangserver-includeIndexing '("**/*.{sv,svh}")
+  "Files included for indexing (glob pattern)"
+  :group 'lsp-svlangserver
+  :type '(repeat string)
+  :safe (lambda (x) (seq-every-p #'stringp x)))
+
+(defcustom lsp-clients-svlangserver-excludeIndexing '("test/**/*.{sv,svh}")
+  "Files excluded for indexing (glob pattern)"
+  :group 'lsp-svlangserver
+  :type '(repeat string)
+  :safe (lambda (x) (seq-every-p #'stringp x)))
+
+(defcustom lsp-clients-svlangserver-defines '()
+  "Defines needed for linting"
+  :group 'lsp-svlangserver
+  :type '(repeat string)
+  :safe (lambda (x) (seq-every-p #'stringp x)))
+
+(defcustom lsp-clients-svlangserver-launchConfiguration "verilator -sv --lint-only -Wall"
+  "Verilator command used for linting"
+  :group 'lsp-svlangserver
+  :type 'string
+  :safe (lambda (x) (stringp x)))
+
+(defcustom lsp-clients-svlangserver-lintOnUnsaved t
+  "Enable linting on unsaved files"
+  :group 'lsp-svlangserver
+  :type 'boolean
+  :safe (lambda (x) (booleanp x)))
+
+(defcustom lsp-clients-svlangserver-formatCommand "verible-verilog-format"
+  "Verible verilog format command"
+  :group 'lsp-svlangserver
+  :type 'string
+  :safe (lambda (x) (stringp x)))
+
+(defcustom lsp-clients-svlangserver-disableCompletionProvider nil
+  "Disable auto completion provided by the language server"
+  :group 'lsp-svlangserver
+  :type 'boolean
+  :safe (lambda (x) (booleanp x)))
+
+(defcustom lsp-clients-svlangserver-disableHoverProvider nil
+  "Disable hover over help provided by the language server"
+  :group 'lsp-svlangserver
+  :type 'boolean
+  :safe (lambda (x) (booleanp x)))
+
+(defcustom lsp-clients-svlangserver-disableSignatureHelpProvider nil
+  "Disable signature help provided by the language server"
+  :group 'lsp-svlangserver
+  :type 'boolean
+  :safe (lambda (x) (booleanp x)))
+
+(defcustom lsp-clients-svlangserver-disableLinting nil
+  "Disable verilator linting"
+  :group 'lsp-svlangserver
+  :type 'boolean
+  :safe (lambda (x) (booleanp x)))
+
+(defun lsp-clients-svlangserver-command ()
+  (list lsp-clients-svlangserver-node-command lsp-clients-svlangserver-module-path "--stdio"))
+
+(defun lsp-clients-svlangserver-get-workspace-additional-dirs (_workspace)
+  lsp-clients-svlangserver-workspace-additional-dirs)
+
+(lsp-register-client
+    (make-lsp-client :new-connection (lsp-stdio-connection 'lsp-clients-svlangserver-command)
+                     :major-modes '(verilog-mode)
+                     :priority -1
```

Comment:
The priority should be different from the priority of `lsp-verilog`.
IMO you can just move this code to `lsp-verilog.el` so we have all language sever for a language in the same file, thus easier to decide the priority there. You will not need the boiler plate comment at file header too.

Reply:
That looks doable. But won't that break it for existing user since they won't have my language server installed?

Reply:
The language server will be decided by priority, in case `lsp-mode` cannot find the language server executable, the current client will be skipped and the next one (has lower priority) will be considered.

Reply:
Ok. Will do. Is there a reference example for this? Specifically how to update the mkdocs.yml so that the Verilog page has entry for both (hdl checker and svlangserver)? Do I need to create it manually under docs/manual-language-docs?
Thanks

Reply:
No, the docs check for the different registered `:group`s, so you can keep it on the same file with different defgroups and it should work

Reply:
You can check `lsp-php.el`